Add ReflectSetter

darinadler opened the pull request:
#### 4c99b6a9d25c8faa8994c277fc98113f6a568b77
<pre>
Add ReflectSetter
<a href="https://rdar.apple.com/146613724">rdar://146613724</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=289440">https://bugs.webkit.org/show_bug.cgi?id=289440</a>

Reviewed by Sam Weinig.

We can cut down on hand-written code by adding [ReflectSetter], like [Reflect],
but only for the set function, not the get function. In addition, did some related
changes:

- removed functions that were previously used to implement setters
- removed uses of these functions, preferring to set attributes with the general
  purpose attribute setting functions rather than calling a named wrapper
- renamed some internal functions to use WebKit naming style rather than DOM, removing &quot;get&quot;
- used &quot;without synchronization&quot; versions of functions more consistently, might lead to
  small performance gains
- addded a numeric attribute setter function so we can use [ReflectSetter] with double
- moved some DOM function implementations out of the headers, not needed for optimization
  and contributes to slower build times

* Source/WebCore/WebCore.xcodeproj/project.pbxproj: Added PopoverInvokerElement.idl
so global searches in Xcode will see things in this file.

* Source/WebCore/accessibility/AccessibilityNodeObject.cpp:
(WebCore::AccessibilityNodeObject::url const): Use getURLAttribute.

* Source/WebCore/accessibility/AccessibilityObject.cpp:
(WebCore::AccessibilityObject::ariaLevel const): Use integralAttribute and std::max.
(WebCore::AccessibilityObject::integralAttribute const): Renamed from getIntegralAttribute.
(WebCore::AccessibilityObject::setSize const): Use integralAttribute.
(WebCore::AccessibilityObject::posInSet const): Use integralAttribute.

(WebCore::AccessibilityObject::getIntegralAttribute const): Deleted.
* Source/WebCore/accessibility/AccessibilityObject.h: Renamed getIntegralAttribute
to integralAttribute.

* Source/WebCore/accessibility/AccessibilityTable.cpp:
(WebCore::AccessibilityTable::isDataTable const): Use integralAttribute.
(WebCore::AccessibilityTable::axColumnCount const): Use integralAttribute.
(WebCore::AccessibilityTable::axRowCount const): Use integralAttribute.

* Source/WebCore/accessibility/AccessibilityTableCell.cpp:
(WebCore::AccessibilityTableCell::axColumnIndex const): Use integralAttribute.
(WebCore::AccessibilityTableCell::axRowIndex const): Use integralAttribute.
(WebCore::AccessibilityTableCell::axColumnSpan const): Use integralAttribute.
(WebCore::AccessibilityTableCell::axRowSpan const): Use integralAttribute.

* Source/WebCore/accessibility/AccessibilityTableRow.cpp:
(WebCore::AccessibilityTableRow::axColumnIndex const): Use integralAttribute.
(WebCore::AccessibilityTableRow::axRowIndex const): Use integralAttribute.

* Source/WebCore/bindings/scripts/CodeGenerator.pm:
(ContentAttributeName): Support ReflectSetter.
(GetterExpression): Pass in &quot;getter&quot;, also use numericAttribute, integralAttribute,
and unsignedIntegralAttribute.
(SetterExpression): Use setNumericAttribute.

* Source/WebCore/bindings/scripts/CodeGeneratorJS.pm:
(NativeToJSValue): Updated for name change to unsignedIntegralAttribute
and integralAttribute.

* Source/WebCore/bindings/scripts/IDLAttributes.json: Added ReflectSetter.

* Source/WebCore/bindings/scripts/test/JS/JSTestObj.cpp: Use integralAttribute.
Added reflected setter tests.

(WebCore::JSTestObjDOMConstructor::construct):
(WebCore::JSTestObjPrototype::finishCreation):
(WebCore::jsTestObj_reflectedIntegralAttrGetter):
(WebCore::jsTestObj_reflectedUnsignedIntegralAttrGetter):
(WebCore::jsTestObj_reflectedCustomIntegralAttrGetter):
(WebCore::JSC_DEFINE_CUSTOM_SETTER):
(WebCore::jsTestObj_reflectedSetterStringAttrGetter):
(WebCore::JSC_DEFINE_CUSTOM_GETTER):
(WebCore::setJSTestObj_reflectedSetterStringAttrSetter):
(WebCore::jsTestObj_reflectedSetterUSVStringAttrGetter):
(WebCore::setJSTestObj_reflectedSetterUSVStringAttrSetter):
(WebCore::jsTestObj_reflectedSetterIntegralAttrGetter):
(WebCore::setJSTestObj_reflectedSetterIntegralAttrSetter):
(WebCore::jsTestObj_reflectedSetterUnsignedIntegralAttrGetter):
(WebCore::setJSTestObj_reflectedSetterUnsignedIntegralAttrSetter):
(WebCore::jsTestObj_reflectedSetterBooleanAttrGetter):
(WebCore::setJSTestObj_reflectedSetterBooleanAttrSetter):
(WebCore::jsTestObj_reflectedSetterElementAttrGetter):
(WebCore::setJSTestObj_reflectedSetterElementAttrSetter):
(WebCore::jsTestObj_reflectedSetterElementsArrayAttrGetter):
(WebCore::setJSTestObj_reflectedSetterElementsArrayAttrSetter):
(WebCore::jsTestObj_reflectedSetterURLAttrGetter):
(WebCore::setJSTestObj_reflectedSetterURLAttrSetter):
(WebCore::jsTestObj_reflectedSetterUSVURLAttrGetter):
(WebCore::setJSTestObj_reflectedSetterUSVURLAttrSetter):
(WebCore::jsTestObj_reflectedSetterCustomIntegralAttrGetter):
(WebCore::setJSTestObj_reflectedSetterCustomIntegralAttrSetter):
(WebCore::jsTestObj_reflectedSetterCustomBooleanAttrGetter):
(WebCore::setJSTestObj_reflectedSetterCustomBooleanAttrSetter):
(WebCore::jsTestObj_reflectedSetterCustomURLAttrGetter):
(WebCore::setJSTestObj_reflectedSetterCustomURLAttrSetter):
* Source/WebCore/bindings/scripts/test/TestObj.idl: Added reflected setter tests.

* Source/WebCore/dom/Document.cpp:
(WebCore::Document::setDir): Use setAttributeWithoutSynchronization.

* Source/WebCore/dom/Element.cpp:
(WebCore::Element::setBooleanAttribute): Use setAttributeWIthoutSynchronization.
(WebCore::Element::integralAttribute const): Renamed from getIntegralAttribute.
Use attributeWithoutSynchronization.
(WebCore::Element::setIntegralAttribute): Use setAttributeWIthoutSynchronization.
(WebCore::Element::unsignedIntegralAttribute const): Renamed from
getUnsignedIntegralAttribute. Use attributeWithoutSynchronization.
(WebCore::Element::setUnsignedIntegralAttribute): Use setAttributeWIthoutSynchronization.
(WebCore::Element::setNumericAttribute): Added.

(WebCore::Element::getIntegralAttribute const): Deleted.
(WebCore::Element::getUnsignedIntegralAttribute const): Deleted.
* Source/WebCore/dom/Element.h: Updated for above changes.

* Source/WebCore/dom/ElementContentEditable.idl: Use ReflectSetter.

* Source/WebCore/dom/ElementInternals.h: Sorted forward declarations.

* Source/WebCore/editing/BreakBlockquoteCommand.cpp:
(WebCore::BreakBlockquoteCommand::doApply): Use setAttributeWithoutSynchronization.
* Source/WebCore/editing/CreateLinkCommand.cpp:
(WebCore::CreateLinkCommand::doApply): Use setAttributeWithoutSynchronization.
* Source/WebCore/editing/Editor.cpp:
(WebCore::Editor::registerAttachmentIdentifier): Use attributeWithoutSynchronization.
* Source/WebCore/editing/EditorCommand.cpp:
(WebCore::executeInsertImage): Use setAttributeWithoutSynchronization.
* Source/WebCore/editing/cocoa/DataDetection.mm:
(WebCore::processDataDetectorScannerResults): Use setAttributeWithoutSynchronization.
* Source/WebCore/editing/cocoa/EditorCocoa.mm:
(WebCore::Editor::insertMultiRepresentationHEIC): Use setAttributeWithoutSynchronization.
* Source/WebCore/html/FileInputType.cpp:
(WebCore::FileInputType::createShadowSubtree): Use setAttributeWithoutSynchronization.

* Source/WebCore/html/HTMLAnchorElement.cpp:
(WebCore::HTMLAnchorElement::setFullURL): Moved this here out of the header.

(WebCore::HTMLAnchorElement::setHref): Deleted.
(WebCore::HTMLAnchorElement::setReferrerPolicyForBindings): Deleted.
* Source/WebCore/html/HTMLAnchorElement.h: Removed setHref
and setReferrerPolicyForBindings, and moved the implementation of setFullURL into
the implementation file.

* Source/WebCore/html/HTMLAnchorElement.idl: Use RelectSetter.

* Source/WebCore/html/HTMLAreaElement.idl: Use RelectSetter.

* Source/WebCore/html/HTMLAttachmentElement.cpp:
(WebCore::HTMLAttachmentElement::updateImage): Use setAttributeWithoutSychronization.
Use removeAttribute.

* Source/WebCore/html/HTMLBaseElement.cpp:
(WebCore::HTMLBaseElement::setHref): Deleted.
* Source/WebCore/html/HTMLBaseElement.h: Deleted setHref.
* Source/WebCore/html/HTMLBaseElement.idl: Use ReflectSetter.

* Source/WebCore/html/HTMLButtonElement.cpp:
(WebCore::HTMLButtonElement::setType): Deleted.
(WebCore::HTMLButtonElement::setCommand): Deleted.
* Source/WebCore/html/HTMLButtonElement.h: Deleted setType and setCommand.
* Source/WebCore/html/HTMLButtonElement.idl: Use ReflectSetter.

* Source/WebCore/html/HTMLDetailsElement.cpp:
(WebCore::HTMLDetailsElement::didAddUserAgentShadowRoot): Use
hasAttributeWithoutSynchronization.
(WebCore::HTMLDetailsElement::ensureDetailsExclusivityAfterMutation):
Use hasAttributeWithoutSynchronization and attributeWithoutSynchronization.
(WebCore::HTMLDetailsElement::toggleOpen): Moved this here out of the header.
* Source/WebCore/html/HTMLDetailsElement.h: Moved toggleOpen out of the header.

* Source/WebCore/html/HTMLDialogElement.cpp:
(WebCore::HTMLDialogElement::show): Use setAttributeWithoutSynchronization.
(WebCore::HTMLDialogElement::showModal): Use setAttributeWithoutSynchronization.
(WebCore::HTMLDialogElement::close): Use removeAttribute.
(WebCore::HTMLDialogElement::isOpen const): Moved this here out of the header.
* Source/WebCore/html/HTMLDialogElement.h: Removed unneeded declaration of
ToggleEventTask and moved isOpen out of the header.

* Source/WebCore/html/HTMLElement.cpp:
(WebCore::runPopoverFocusingSteps): Use hasAttributeWithoutSynchronization.

(WebCore::HTMLElement::setDir): Deleted.
(WebCore::HTMLElement::setAutocapitalize): Deleted.
(WebCore::HTMLElement::setInputMode): Deleted.
(WebCore::HTMLElement::setEnterKeyHint): Deleted.
* Source/WebCore/html/HTMLElement.h: Removed setDir, setInputMode, setAutocapitalize,
and setEnterKeyHint.
* Source/WebCore/html/HTMLElement.idl: Use ReflectSetter.

* Source/WebCore/html/HTMLFormControlElement.cpp:
(WebCore::HTMLFormControlElement::setFormEnctype): Deleted.
(WebCore::HTMLFormControlElement::setFormMethod): Deleted.
(WebCore::HTMLFormControlElement::setFormAction): Deleted.
(WebCore::HTMLFormControlElement::setAutocomplete): Deleted.
(WebCore::HTMLFormControlElement::setPopoverTargetAction): Deleted.
* Source/WebCore/html/HTMLFormControlElement.h: Removed setFormEnctype, setFormMethod,
and setFormAction.

* Source/WebCore/html/HTMLFormElement.cpp:
(WebCore::HTMLFormElement::setAction): Deleted.
(WebCore::HTMLFormElement::setEnctype): Deleted.
(WebCore::HTMLFormElement::setMethod): Deleted.
(WebCore::HTMLFormElement::setAutocomplete): Deleted.
* Source/WebCore/html/HTMLFormElement.h: Removed setEnctype, setAutocomplete, setAction,
and setMethod.
* Source/WebCore/html/HTMLFormElement.idl: Use ReflectSetter.

* Source/WebCore/html/HTMLIFrameElement.cpp:
(WebCore::HTMLIFrameElement::setReferrerPolicyForBindings): Deleted.
(WebCore::HTMLIFrameElement::setLoading): Deleted.
* Source/WebCore/html/HTMLIFrameElement.h: Removed setReferrerPolicyForBindings and
setLoading.
* Source/WebCore/html/HTMLIFrameElement.idl: Use ReflectSetter.

* Source/WebCore/html/HTMLImageElement.cpp:
(WebCore::HTMLImageElement::createForLegacyFactoryFunction): Use
setUnsignedIntegralAttribute.
(WebCore::HTMLImageElement::alt const): Deleted.
(WebCore::HTMLImageElement::setHeight): Deleted.
(WebCore::HTMLImageElement::src const): Deleted.
(WebCore::HTMLImageElement::setSrc): Deleted.
(WebCore::HTMLImageElement::setWidth): Deleted.
(WebCore::HTMLImageElement::setDecoding): Deleted.
(WebCore::HTMLImageElement::setCrossOrigin): Deleted.
(WebCore::HTMLImageElement::setSrcsetForBindings): Deleted.
(WebCore::HTMLImageElement::setLoadingForBindings): Deleted.
(WebCore::HTMLImageElement::setReferrerPolicyForBindings): Deleted.
(WebCore::HTMLImageElement::setFetchPriorityForBindings): Deleted.
* Source/WebCore/html/HTMLImageElement.h: Removed alt, setHeight, src, setSrc,
setCrossOrigin, setWidth, setDecodeing, setSrcsetForBindings, setLoadingForBindings,
setReferrerPolicyForBindings, and setFetchPriorityForBindings.
* Source/WebCore/html/HTMLImageElement.idl: Use Reflect for more cases where it works.
Use ReflectSetter.

* Source/WebCore/html/HTMLInputElement.cpp:
(WebCore::HTMLInputElement::needsSuspensionCallback): Use atttributeWithoutSynchronization.
(WebCore::HTMLInputElement::addSubresourceAttributeURLs const): Use getURLAttribute.
(WebCore::HTMLInputElement::setType): Deleted.
(WebCore::HTMLInputElement::defaultValue const): Deleted.
(WebCore::HTMLInputElement::setDefaultValue): Deleted.
(WebCore::HTMLInputElement::alt const): Deleted.
(WebCore::HTMLInputElement::src const): Deleted.
(WebCore::HTMLInputElement::setHeight): Deleted.
(WebCore::HTMLInputElement::setWidth): Deleted.
* Source/WebCore/html/HTMLInputElement.h: Removed setHeight, defaultValue, setDefaultValue,
setType, setWidth, alt, and src.
* Source/WebCore/html/HTMLInputElement.idl: Use ReflectSetter.

* Source/WebCore/html/HTMLLinkElement.cpp:
(WebCore::HTMLLinkElement::setCrossOrigin): Deleted.
(WebCore::HTMLLinkElement::setAs): Deleted.
(WebCore::HTMLLinkElement::setReferrerPolicyForBindings): Deleted.
(WebCore::HTMLLinkElement::setFetchPriorityForBindings): Deleted.
* Source/WebCore/html/HTMLLinkElement.h: Removed setCrossOrigin, setAs,
setReferrerPolicyForBindings, and setFetchPriorityForBindings.
* Source/WebCore/html/HTMLLinkElement.idl: Use ReflectSetter.

* Source/WebCore/html/HTMLMarqueeElement.cpp:
(WebCore::HTMLMarqueeElement::loop const): Use integralAttribute.

* Source/WebCore/html/HTMLMediaElement.cpp:
(WebCore::HTMLMediaElement::outOfBandTrackSources): Use attributeWithoutSynchronization.
(WebCore::HTMLMediaElement::setCrossOrigin): Deleted.
* Source/WebCore/html/HTMLMediaElement.h: Removed setCrossOrigin.
* Source/WebCore/html/HTMLMediaElement.idl: Use ReflectSetter.

* Source/WebCore/html/HTMLMeterElement.cpp:
(WebCore::HTMLMeterElement::setMin): Deleted.
(WebCore::HTMLMeterElement::setMax): Deleted.
(WebCore::HTMLMeterElement::setValue): Deleted.
(WebCore::HTMLMeterElement::setLow): Deleted.
(WebCore::HTMLMeterElement::setHigh): Deleted.
(WebCore::HTMLMeterElement::setOptimum): Deleted.
* Source/WebCore/html/HTMLMeterElement.h: Removed setMin, setMax, setValue, setLow,
setHigh, and setOptimum.
* Source/WebCore/html/HTMLMeterElement.idl: Use ReflectSetter.

* Source/WebCore/html/HTMLOListElement.cpp:
(WebCore::HTMLOListElement::setStartForBindings): Deleted.
* Source/WebCore/html/HTMLOListElement.h: Removed setStartForBindings.
* Source/WebCore/html/HTMLOListElement.idl: Use ReflectSetter.

* Source/WebCore/html/HTMLOptionElement.cpp:
(WebCore::HTMLOptionElement::createForLegacyFactoryFunction):
Use setAttributeWithoutSynchronization.
(WebCore::HTMLOptionElement::value const):
(WebCore::HTMLOptionElement::label const):
(WebCore::HTMLOptionElement::displayLabel const):
(WebCore::HTMLOptionElement::collectOptionInnerTextCollapsingWhitespace const):
(WebCore::HTMLOptionElement::setValue): Deleted.
(WebCore::HTMLOptionElement::setLabel): Deleted.
* Source/WebCore/html/HTMLOptionElement.h: Removed setValue and setLabel.
* Source/WebCore/html/HTMLOptionElement.idl: Use ReflectSetter.

* Source/WebCore/html/HTMLProgressElement.cpp:
(WebCore::HTMLProgressElement::setValue): Deleted.
* Source/WebCore/html/HTMLProgressElement.h: Removed setValue.
* Source/WebCore/html/HTMLProgressElement.idl: Use ReflectSetter.

* Source/WebCore/html/HTMLScriptElement.cpp:
(WebCore::HTMLScriptElement::setCrossOrigin): Deleted.
(WebCore::HTMLScriptElement::setReferrerPolicyForBindings): Deleted.
(WebCore::HTMLScriptElement::setFetchPriorityForBindings): Deleted.
* Source/WebCore/html/HTMLScriptElement.h: Removed setCrossOrigin,
setReferrerPolicyForBindings, and setFetchPriorityForBindings.
* Source/WebCore/html/HTMLScriptElement.idl: Use ReflectSetter.

* Source/WebCore/html/HTMLSelectElement.idl: Use ReflectSetter.

* Source/WebCore/html/HTMLTableCellElement.cpp:
(WebCore::HTMLTableCellElement::setScope): Deleted.
* Source/WebCore/html/HTMLTableCellElement.h: Removed setScope.
* Source/WebCore/html/HTMLTableCellElement.idl: Use ReflectSetter.

* Source/WebCore/html/HTMLTemplateElement.cpp:
(WebCore::HTMLTemplateElement::setShadowRootMode): Deleted.
* Source/WebCore/html/HTMLTemplateElement.h: Removed setShadowRootMode.
* Source/WebCore/html/HTMLTemplateElement.idl: Use ReflectSetter.

* Source/WebCore/html/HTMLTextAreaElement.idl: Use ReflectSetter.

* Source/WebCore/html/HTMLTrackElement.cpp:
(WebCore::HTMLTrackElement::HTMLTrackElement): Replaced pointless code that
called attributeWithoutSynchronization in the constructor. There are no
attributes yet at construction time, so this is just a slow way to get what
is always a null string.
(WebCore::HTMLTrackElement::setKind): Deleted.
(WebCore::HTMLTrackElement::srclang const): Deleted.
(WebCore::HTMLTrackElement::label const): Deleted.
* Source/WebCore/html/HTMLTrackElement.h: Deleted setKind, srclang, and label.
* Source/WebCore/html/HTMLTrackElement.idl: Use ReflectSetter.

* Source/WebCore/html/HTMLVideoElement.cpp:
(WebCore::HTMLVideoElement::setWebkitWirelessVideoPlaybackDisabled): Deleted.
* Source/WebCore/html/HTMLVideoElement.h: Removed setWebkitWirelessVideoPlaybackDisabled.
* Source/WebCore/html/HTMLVideoElement.idl: Use ReflectSetter.

* Source/WebCore/html/ImageDocument.cpp:
(WebCore::ImageDocument::createDocumentStructure): Use setAttributeWithoutSynchronization.
(WebCore::ImageDocument::resizeImageToFit): Use setIntegralAttribute.
(WebCore::ImageDocument::restoreImageSize): Use setIntegralAttribute.

* Source/WebCore/html/InputType.cpp:
(WebCore::InputType::saveFormControlState const): Use attributeWithoutSynchronization.

* Source/WebCore/html/PermissionsPolicy.cpp:
(WebCore::PermissionsPolicy::processPermissionsPolicyAttribute): Use
hasAttributeWithoutSynchronization.

* Source/WebCore/html/PopoverInvokerElement.idl: Use ReflectSetter.

* Source/WebCore/html/shadow/DateTimeNumericFieldElement.cpp:
(WebCore::DateTimeNumericFieldElement::setARIAValueAttributesWithInteger):
Use a single call to AtomString::number for better efficiency.

* Source/WebCore/loader/ImageLoader.cpp:
(WebCore::ImageLoader::updateFromElement): Use getURLAttribute.

* Source/WebCore/mathml/MathMLMencloseElement.cpp:
(WebCore::MathMLMencloseElement::parseNotationAttribute): Use
hasAttributeWithoutSynchronization.

* Source/WebCore/mathml/MathMLSelectElement.cpp:
(WebCore::MathMLSelectElement::getSelectedActionChildAndIndex): Use
integralAttribute.

* Source/WebCore/page/DragController.cpp:
(WebCore::documentFragmentFromDragData): Use setAttributeWithoutSynchronization.

* Source/WebCore/page/text-extraction/TextExtraction.cpp:
(WebCore::TextExtraction::extractItemData): Use getURLAttribute.

* Source/WebCore/rendering/HitTestResult.cpp:
(WebCore::HitTestResult::altDisplayString const): Use attributeWithoutSynchronization.

* Source/WebKitLegacy/mac/DOM/DOMHTMLBaseElement.mm:
(-[DOMHTMLBaseElement setHref:]): Use setAttributeWithoutSynchronization.
* Source/WebKitLegacy/mac/DOM/DOMHTMLButtonElement.mm:
(-[DOMHTMLButtonElement setType:]): Use setAttributeWithoutSynchronization.
* Source/WebKitLegacy/mac/DOM/DOMHTMLElement.mm:
(-[DOMHTMLElement setDir:]): Use setAttributeWithoutSynchronization.
(-[DOMHTMLElement setAutocapitalize:]): Use setAttributeWithoutSynchronization.
* Source/WebKitLegacy/mac/DOM/DOMHTMLEmbedElement.mm:
(-[DOMHTMLEmbedElement height]): Use integralAttribute.
(-[DOMHTMLEmbedElement width]): Use integralAttribute.
* Source/WebKitLegacy/mac/DOM/DOMHTMLFormElement.mm:
(-[DOMHTMLFormElement setAutocomplete:]): Use setAttributeWithoutSynchronization.
(-[DOMHTMLFormElement setEnctype:]): Use setAttributeWithoutSynchronization.
(-[DOMHTMLFormElement setEncoding:]): Use setAttributeWithoutSynchronization.
(-[DOMHTMLFormElement setMethod:]): Use setAttributeWithoutSynchronization.
* Source/WebKitLegacy/mac/DOM/DOMHTMLImageElement.mm:
(-[DOMHTMLImageElement setCrossOrigin:]): Use setAttributeWithoutSynchronization.
(-[DOMHTMLImageElement setHeight:]): Use setAttributeWithoutSynchronization.
(-[DOMHTMLImageElement hspace]): Use integralAttribute.
(-[DOMHTMLImageElement vspace]): Use integralAttribute.
(-[DOMHTMLImageElement setWidth:]): Use setIntegralAttribute.
(-[DOMHTMLImageElement altDisplayString]): Use attributeWithoutSynchronization.
* Source/WebKitLegacy/mac/DOM/DOMHTMLInputElement.mm:
(-[DOMHTMLInputElement setAutocomplete:]): Use setAttributeWithoutSynchronization.
(-[DOMHTMLInputElement setFormAction:]): Use setAttributeWithoutSynchronization.
(-[DOMHTMLInputElement setFormEnctype:]): Use setAttributeWithoutSynchronization.
(-[DOMHTMLInputElement setFormMethod:]): Use setAttributeWithoutSynchronization.
(-[DOMHTMLInputElement setHeight:]): Use setUnsignedIntegralAttribute.
(-[DOMHTMLInputElement max]): Use attributeWithoutSynchronization.
(-[DOMHTMLInputElement min]): Use attributeWithoutSynchronization.
(-[DOMHTMLInputElement pattern]): Use attributeWithoutSynchronization.
(-[DOMHTMLInputElement placeholder]): Use attributeWithoutSynchronization.
(-[DOMHTMLInputElement step]): Use attributeWithoutSynchronization.
(-[DOMHTMLInputElement setType:]): Use setAttributeWithoutSynchronization.
(-[DOMHTMLInputElement defaultValue]): Use attributeWithoutSynchronization.
(-[DOMHTMLInputElement setDefaultValue:]): Use setAttributeWithoutSynchronization.
(-[DOMHTMLInputElement setWidth:]): Use setUnsignedIntegralAttribute.
(-[DOMHTMLInputElement altDisplayString]): Use attributeWithoutSynchronization.
* Source/WebKitLegacy/mac/DOM/DOMHTMLLIElement.mm:
(-[DOMHTMLLIElement value]): Use integralAttribute.
* Source/WebKitLegacy/mac/DOM/DOMHTMLLinkElement.mm:
(-[DOMHTMLLinkElement setCrossOrigin:]): Use setAttributeWithoutSynchronization.
* Source/WebKitLegacy/mac/DOM/DOMHTMLMarqueeElement.mm:
(-[DOMHTMLMarqueeElement hspace]): Use unsignedIntegralAttribute.
(-[DOMHTMLMarqueeElement vspace]): Use unsignedIntegralAttribute.
* Source/WebKitLegacy/mac/DOM/DOMHTMLMediaElement.mm:
(-[DOMHTMLMediaElement setCrossOrigin:]): Use setAttributeWithoutSynchronization.
* Source/WebKitLegacy/mac/DOM/DOMHTMLOListElement.mm:
(-[DOMHTMLOListElement setStart:]): Use setIntegralAttribute.
* Source/WebKitLegacy/mac/DOM/DOMHTMLObjectElement.mm:
(-[DOMHTMLObjectElement hspace]): Use integralAttribute.
(-[DOMHTMLObjectElement vspace]): Use integralAttribute.
* Source/WebKitLegacy/mac/DOM/DOMHTMLOptionElement.mm:
(-[DOMHTMLOptionElement setLabel:]): Use setAttributeWithoutSynchronization.
(-[DOMHTMLOptionElement setValue:]): Use setAttributeWithoutSynchronization.
* Source/WebKitLegacy/mac/DOM/DOMHTMLPreElement.mm:
(-[DOMHTMLPreElement width]): Use integralAttribute.
* Source/WebKitLegacy/mac/DOM/DOMHTMLScriptElement.mm:
(-[DOMHTMLScriptElement htmlFor]): Use attributeWithoutSynchronization.
(-[DOMHTMLScriptElement event]): Use attributeWithoutSynchronization.
(-[DOMHTMLScriptElement charset]): Use attributeWithoutSynchronization.
(-[DOMHTMLScriptElement type]): Use attributeWithoutSynchronization.
(-[DOMHTMLScriptElement setCrossOrigin:]): Use setAttributeWithoutSynchronization.
(-[DOMHTMLScriptElement nonce]): Use attributeWithoutSynchronization.
* Source/WebKitLegacy/mac/DOM/DOMHTMLTableCellElement.mm:
(-[DOMHTMLTableCellElement setScope:]): Use setAttributeWithoutSynchronization.
* Source/WebKitLegacy/mac/DOM/DOMHTMLTextAreaElement.mm:
(-[DOMHTMLTextAreaElement setAutocomplete:]): Use setAttributeWithoutSynchronization.
* Source/WebKitLegacy/mac/DOM/DOMHTMLVideoElement.mm:
(-[DOMHTMLVideoElement width]): Use unsignedIntegralAttribute.
(-[DOMHTMLVideoElement height]): Use unsignedIntegralAttribute.

Canonical link: <a href="https://commits.webkit.org/296427@main">https://commits.webkit.org/296427@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/fc4391d18cf6b17c99ae2b940532bc51ad35864a

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/108501 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/28162 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/18586 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/113710 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/58900 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/110464 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/28851 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/36714 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/82400 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/111449 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/22888 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/97729 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/62836 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/22306 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/15867 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/58426 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/92259 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/15920 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/116831 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/35555 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/26201 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/91425 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/35928 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/94003 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/91226 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/23243 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/36121 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/13884 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/31302 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/35457 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/40992 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/35166 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/38518 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/36847 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/62335a1ba70966a4e837e9c7c03c8e3fe41c3bb1. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/45901

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/09817b37a60421da6c998fcb515d6821803da7e1. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/45901

weinig commented on Source/WebCore/page/text-extraction/TextExtraction.cpp:
```diff
@@ -316,7 +316,7 @@ static inline Variant<SkipExtraction, ItemData, URL, Editable> extractItemData(N
 
     if (RefPtr image = dynamicDowncast<HTMLImageElement>(element))
-        return { ImageItemData { image->src().lastPathComponent().toString(), image->altText() } };
+        return { ImageItemData { image->getURLAttribute(HTMLNames::srcAttr).lastPathComponent().toString(), image->altText() } };
```
Are you leaving the `get` prefix in this one so that URL stays capitalized ;)?

darinadler replied:
You know me too well. That was one reason, but another was that I was doing too much renaming in this patch and decided to cool it. I probably would have renamed it to something that didn’t start with “url” if my primary mission was renaming, but the renaming was a distraction.

weinig reviewed (approved):
Excited to see you are back in the writing PR game!

"+247 −604" nice ratio!

`[ReflectSetter]` could use a bindings test. Source/WebCore/bindings/scripts/test/TestObj.idl already has some `[Reflect]` tests, so that would be a good file for them.

darinadler commented:
> `[ReflectSetter]` could use a bindings test. Source/WebCore/bindings/scripts/test/TestObj.idl already has some `[Reflect]` tests, so that would be a good file for them.

Makes sense. I should probably add multiple bindings tests for the different types we can reflect.

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/4373ea14487d7ac04302e3e7ca1c8ce7e9c45741. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/45901

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/cd3fd04c016c7681eb3092d0be8f8d30038a2dcc. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/45901

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/c5bf722f3ce650b5a678eb6fe92577c4d142a0ec. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/45901

darinadler committed:
Add ReflectSetter
rdar://146613724
https://bugs.webkit.org/show_bug.cgi?id=289440

Reviewed by Sam Weinig.

We can cut down on hand-written code by adding [ReflectSetter], like [Reflect],
but only for the set function, not the get function. In addition, did some related
changes:

- removed functions that were previously used to implement setters
- removed uses of these functions, preferring to set attributes with the general
  purpose attribute setting functions rather than calling a named wrapper
- renamed some internal functions to use WebKit naming style rather than DOM, removing "get"
- used "without synchronization" versions of functions more consistently, might lead to
  small performance gains
- addded a numeric attribute setter function so we can use [ReflectSetter] with double
- moved some DOM function implementations out of the headers, not needed for optimization
  and contributes to slower build times

* Source/WebCore/WebCore.xcodeproj/project.pbxproj: Added PopoverInvokerElement.idl
so global searches in Xcode will see things in this file.

* Source/WebCore/accessibility/AccessibilityNodeObject.cpp:
(WebCore::AccessibilityNodeObject::url const): Use getURLAttribute.

* Source/WebCore/accessibility/AccessibilityObject.cpp:
(WebCore::AccessibilityObject::ariaLevel const): Use integralAttribute and std::max.
(WebCore::AccessibilityObject::integralAttribute const): Renamed from getIntegralAttribute.
(WebCore::AccessibilityObject::setSize const): Use integralAttribute.
(WebCore::AccessibilityObject::posInSet const): Use integralAttribute.

(WebCore::AccessibilityObject::getIntegralAttribute const): Deleted.
* Source/WebCore/accessibility/AccessibilityObject.h: Renamed getIntegralAttribute
to integralAttribute.

* Source/WebCore/accessibility/AccessibilityTable.cpp:
(WebCore::AccessibilityTable::isDataTable const): Use integralAttribute.
(WebCore::AccessibilityTable::axColumnCount const): Use integralAttribute.
(WebCore::AccessibilityTable::axRowCount const): Use integralAttribute.

* Source/WebCore/accessibility/AccessibilityTableCell.cpp:
(WebCore::AccessibilityTableCell::axColumnIndex const): Use integralAttribute.
(WebCore::AccessibilityTableCell::axRowIndex const): Use integralAttribute.
(WebCore::AccessibilityTableCell::axColumnSpan const): Use integralAttribute.
(WebCore::AccessibilityTableCell::axRowSpan const): Use integralAttribute.

* Source/WebCore/accessibility/AccessibilityTableRow.cpp:
(WebCore::AccessibilityTableRow::axColumnIndex const): Use integralAttribute.
(WebCore::AccessibilityTableRow::axRowIndex const): Use integralAttribute.

* Source/WebCore/bindings/scripts/CodeGenerator.pm:
(ContentAttributeName): Support ReflectSetter.
(GetterExpression): Pass in "getter", also use numericAttribute, integralAttribute,
and unsignedIntegralAttribute.
(SetterExpression): Use setNumericAttribute.

* Source/WebCore/bindings/scripts/CodeGeneratorJS.pm:
(NativeToJSValue): Updated for name change to unsignedIntegralAttribute
and integralAttribute.

* Source/WebCore/bindings/scripts/IDLAttributes.json: Added ReflectSetter.

* Source/WebCore/bindings/scripts/test/JS/JSTestObj.cpp: Use integralAttribute.
Added reflected setter tests.

(WebCore::JSTestObjDOMConstructor::construct):
(WebCore::JSTestObjPrototype::finishCreation):
(WebCore::jsTestObj_reflectedIntegralAttrGetter):
(WebCore::jsTestObj_reflectedUnsignedIntegralAttrGetter):
(WebCore::jsTestObj_reflectedCustomIntegralAttrGetter):
(WebCore::JSC_DEFINE_CUSTOM_SETTER):
(WebCore::jsTestObj_reflectedSetterStringAttrGetter):
(WebCore::JSC_DEFINE_CUSTOM_GETTER):
(WebCore::setJSTestObj_reflectedSetterStringAttrSetter):
(WebCore::jsTestObj_reflectedSetterUSVStringAttrGetter):
(WebCore::setJSTestObj_reflectedSetterUSVStringAttrSetter):
(WebCore::jsTestObj_reflectedSetterIntegralAttrGetter):
(WebCore::setJSTestObj_reflectedSetterIntegralAttrSetter):
(WebCore::jsTestObj_reflectedSetterUnsignedIntegralAttrGetter):
(WebCore::setJSTestObj_reflectedSetterUnsignedIntegralAttrSetter):
(WebCore::jsTestObj_reflectedSetterBooleanAttrGetter):
(WebCore::setJSTestObj_reflectedSetterBooleanAttrSetter):
(WebCore::jsTestObj_reflectedSetterElementAttrGetter):
(WebCore::setJSTestObj_reflectedSetterElementAttrSetter):
(WebCore::jsTestObj_reflectedSetterElementsArrayAttrGetter):
(WebCore::setJSTestObj_reflectedSetterElementsArrayAttrSetter):
(WebCore::jsTestObj_reflectedSetterURLAttrGetter):
(WebCore::setJSTestObj_reflectedSetterURLAttrSetter):
(WebCore::jsTestObj_reflectedSetterUSVURLAttrGetter):
(WebCore::setJSTestObj_reflectedSetterUSVURLAttrSetter):
(WebCore::jsTestObj_reflectedSetterCustomIntegralAttrGetter):
(WebCore::setJSTestObj_reflectedSetterCustomIntegralAttrSetter):
(WebCore::jsTestObj_reflectedSetterCustomBooleanAttrGetter):
(WebCore::setJSTestObj_reflectedSetterCustomBooleanAttrSetter):
(WebCore::jsTestObj_reflectedSetterCustomURLAttrGetter):
(WebCore::setJSTestObj_reflectedSetterCustomURLAttrSetter):
* Source/WebCore/bindings/scripts/test/TestObj.idl: Added reflected setter tests.

* Source/WebCore/dom/Document.cpp:
(WebCore::Document::setDir): Use setAttributeWithoutSynchronization.

* Source/WebCore/dom/Element.cpp:
(WebCore::Element::setBooleanAttribute): Use setAttributeWIthoutSynchronization.
(WebCore::Element::integralAttribute const): Renamed from getIntegralAttribute.
Use attributeWithoutSynchronization.
(WebCore::Element::setIntegralAttribute): Use setAttributeWIthoutSynchronization.
(WebCore::Element::unsignedIntegralAttribute const): Renamed from
getUnsignedIntegralAttribute. Use attributeWithoutSynchronization.
(WebCore::Element::setUnsignedIntegralAttribute): Use setAttributeWIthoutSynchronization.
(WebCore::Element::setNumericAttribute): Added.

(WebCore::Element::getIntegralAttribute const): Deleted.
(WebCore::Element::getUnsignedIntegralAttribute const): Deleted.
* Source/WebCore/dom/Element.h: Updated for above changes.

* Source/WebCore/dom/ElementContentEditable.idl: Use ReflectSetter.

* Source/WebCore/dom/ElementInternals.h: Sorted forward declarations.

* Source/WebCore/editing/BreakBlockquoteCommand.cpp:
(WebCore::BreakBlockquoteCommand::doApply): Use setAttributeWithoutSynchronization.
* Source/WebCore/editing/CreateLinkCommand.cpp:
(WebCore::CreateLinkCommand::doApply): Use setAttributeWithoutSynchronization.
* Source/WebCore/editing/Editor.cpp:
(WebCore::Editor::registerAttachmentIdentifier): Use attributeWithoutSynchronization.
* Source/WebCore/editing/EditorCommand.cpp:
(WebCore::executeInsertImage): Use setAttributeWithoutSynchronization.
* Source/WebCore/editing/cocoa/DataDetection.mm:
(WebCore::processDataDetectorScannerResults): Use setAttributeWithoutSynchronization.
* Source/WebCore/editing/cocoa/EditorCocoa.mm:
(WebCore::Editor::insertMultiRepresentationHEIC): Use setAttributeWithoutSynchronization.
* Source/WebCore/html/FileInputType.cpp:
(WebCore::FileInputType::createShadowSubtree): Use setAttributeWithoutSynchronization.

* Source/WebCore/html/HTMLAnchorElement.cpp:
(WebCore::HTMLAnchorElement::setFullURL): Moved this here out of the header.

(WebCore::HTMLAnchorElement::setHref): Deleted.
(WebCore::HTMLAnchorElement::setReferrerPolicyForBindings): Deleted.
* Source/WebCore/html/HTMLAnchorElement.h: Removed setHref
and setReferrerPolicyForBindings, and moved the implementation of setFullURL into
the implementation file.

* Source/WebCore/html/HTMLAnchorElement.idl: Use RelectSetter.

* Source/WebCore/html/HTMLAreaElement.idl: Use RelectSetter.

* Source/WebCore/html/HTMLAttachmentElement.cpp:
(WebCore::HTMLAttachmentElement::updateImage): Use setAttributeWithoutSychronization.
Use removeAttribute.

* Source/WebCore/html/HTMLBaseElement.cpp:
(WebCore::HTMLBaseElement::setHref): Deleted.
* Source/WebCore/html/HTMLBaseElement.h: Deleted setHref.
* Source/WebCore/html/HTMLBaseElement.idl: Use ReflectSetter.

* Source/WebCore/html/HTMLButtonElement.cpp:
(WebCore::HTMLButtonElement::setType): Deleted.
(WebCore::HTMLButtonElement::setCommand): Deleted.
* Source/WebCore/html/HTMLButtonElement.h: Deleted setType and setCommand.
* Source/WebCore/html/HTMLButtonElement.idl: Use ReflectSetter.

* Source/WebCore/html/HTMLDetailsElement.cpp:
(WebCore::HTMLDetailsElement::didAddUserAgentShadowRoot): Use
hasAttributeWithoutSynchronization.
(WebCore::HTMLDetailsElement::ensureDetailsExclusivityAfterMutation):
Use hasAttributeWithoutSynchronization and attributeWithoutSynchronization.
(WebCore::HTMLDetailsElement::toggleOpen): Moved this here out of the header.
* Source/WebCore/html/HTMLDetailsElement.h: Moved toggleOpen out of the header.

* Source/WebCore/html/HTMLDialogElement.cpp:
(WebCore::HTMLDialogElement::show): Use setAttributeWithoutSynchronization.
(WebCore::HTMLDialogElement::showModal): Use setAttributeWithoutSynchronization.
(WebCore::HTMLDialogElement::close): Use removeAttribute.
(WebCore::HTMLDialogElement::isOpen const): Moved this here out of the header.
* Source/WebCore/html/HTMLDialogElement.h: Removed unneeded declaration of
ToggleEventTask and moved isOpen out of the header.

* Source/WebCore/html/HTMLElement.cpp:
(WebCore::runPopoverFocusingSteps): Use hasAttributeWithoutSynchronization.

(WebCore::HTMLElement::setDir): Deleted.
(WebCore::HTMLElement::setAutocapitalize): Deleted.
(WebCore::HTMLElement::setInputMode): Deleted.
(WebCore::HTMLElement::setEnterKeyHint): Deleted.
* Source/WebCore/html/HTMLElement.h: Removed setDir, setInputMode, setAutocapitalize,
and setEnterKeyHint.
* Source/WebCore/html/HTMLElement.idl: Use ReflectSetter.

* Source/WebCore/html/HTMLFormControlElement.cpp:
(WebCore::HTMLFormControlElement::setFormEnctype): Deleted.
(WebCore::HTMLFormControlElement::setFormMethod): Deleted.
(WebCore::HTMLFormControlElement::setFormAction): Deleted.
(WebCore::HTMLFormControlElement::setAutocomplete): Deleted.
(WebCore::HTMLFormControlElement::setPopoverTargetAction): Deleted.
* Source/WebCore/html/HTMLFormControlElement.h: Removed setFormEnctype, setFormMethod,
and setFormAction.

* Source/WebCore/html/HTMLFormElement.cpp:
(WebCore::HTMLFormElement::setAction): Deleted.
(WebCore::HTMLFormElement::setEnctype): Deleted.
(WebCore::HTMLFormElement::setMethod): Deleted.
(WebCore::HTMLFormElement::setAutocomplete): Deleted.
* Source/WebCore/html/HTMLFormElement.h: Removed setEnctype, setAutocomplete, setAction,
and setMethod.
* Source/WebCore/html/HTMLFormElement.idl: Use ReflectSetter.

* Source/WebCore/html/HTMLIFrameElement.cpp:
(WebCore::HTMLIFrameElement::setReferrerPolicyForBindings): Deleted.
(WebCore::HTMLIFrameElement::setLoading): Deleted.
* Source/WebCore/html/HTMLIFrameElement.h: Removed setReferrerPolicyForBindings and
setLoading.
* Source/WebCore/html/HTMLIFrameElement.idl: Use ReflectSetter.

* Source/WebCore/html/HTMLImageElement.cpp:
(WebCore::HTMLImageElement::createForLegacyFactoryFunction): Use
setUnsignedIntegralAttribute.
(WebCore::HTMLImageElement::alt const): Deleted.
(WebCore::HTMLImageElement::setHeight): Deleted.
(WebCore::HTMLImageElement::src const): Deleted.
(WebCore::HTMLImageElement::setSrc): Deleted.
(WebCore::HTMLImageElement::setWidth): Deleted.
(WebCore::HTMLImageElement::setDecoding): Deleted.
(WebCore::HTMLImageElement::setCrossOrigin): Deleted.
(WebCore::HTMLImageElement::setSrcsetForBindings): Deleted.
(WebCore::HTMLImageElement::setLoadingForBindings): Deleted.
(WebCore::HTMLImageElement::setReferrerPolicyForBindings): Deleted.
(WebCore::HTMLImageElement::setFetchPriorityForBindings): Deleted.
* Source/WebCore/html/HTMLImageElement.h: Removed alt, setHeight, src, setSrc,
setCrossOrigin, setWidth, setDecodeing, setSrcsetForBindings, setLoadingForBindings,
setReferrerPolicyForBindings, and setFetchPriorityForBindings.
* Source/WebCore/html/HTMLImageElement.idl: Use Reflect for more cases where it works.
Use ReflectSetter.

* Source/WebCore/html/HTMLInputElement.cpp:
(WebCore::HTMLInputElement::needsSuspensionCallback): Use atttributeWithoutSynchronization.
(WebCore::HTMLInputElement::addSubresourceAttributeURLs const): Use getURLAttribute.
(WebCore::HTMLInputElement::setType): Deleted.
(WebCore::HTMLInputElement::defaultValue const): Deleted.
(WebCore::HTMLInputElement::setDefaultValue): Deleted.
(WebCore::HTMLInputElement::alt const): Deleted.
(WebCore::HTMLInputElement::src const): Deleted.
(WebCore::HTMLInputElement::setHeight): Deleted.
(WebCore::HTMLInputElement::setWidth): Deleted.
* Source/WebCore/html/HTMLInputElement.h: Removed setHeight, defaultValue, setDefaultValue,
setType, setWidth, alt, and src.
* Source/WebCore/html/HTMLInputElement.idl: Use ReflectSetter.

* Source/WebCore/html/HTMLLinkElement.cpp:
(WebCore::HTMLLinkElement::setCrossOrigin): Deleted.
(WebCore::HTMLLinkElement::setAs): Deleted.
(WebCore::HTMLLinkElement::setReferrerPolicyForBindings): Deleted.
(WebCore::HTMLLinkElement::setFetchPriorityForBindings): Deleted.
* Source/WebCore/html/HTMLLinkElement.h: Removed setCrossOrigin, setAs,
setReferrerPolicyForBindings, and setFetchPriorityForBindings.
* Source/WebCore/html/HTMLLinkElement.idl: Use ReflectSetter.

* Source/WebCore/html/HTMLMarqueeElement.cpp:
(WebCore::HTMLMarqueeElement::loop const): Use integralAttribute.

* Source/WebCore/html/HTMLMediaElement.cpp:
(WebCore::HTMLMediaElement::outOfBandTrackSources): Use attributeWithoutSynchronization.
(WebCore::HTMLMediaElement::setCrossOrigin): Deleted.
* Source/WebCore/html/HTMLMediaElement.h: Removed setCrossOrigin.
* Source/WebCore/html/HTMLMediaElement.idl: Use ReflectSetter.

* Source/WebCore/html/HTMLMeterElement.cpp:
(WebCore::HTMLMeterElement::setMin): Deleted.
(WebCore::HTMLMeterElement::setMax): Deleted.
(WebCore::HTMLMeterElement::setValue): Deleted.
(WebCore::HTMLMeterElement::setLow): Deleted.
(WebCore::HTMLMeterElement::setHigh): Deleted.
(WebCore::HTMLMeterElement::setOptimum): Deleted.
* Source/WebCore/html/HTMLMeterElement.h: Removed setMin, setMax, setValue, setLow,
setHigh, and setOptimum.
* Source/WebCore/html/HTMLMeterElement.idl: Use ReflectSetter.

* Source/WebCore/html/HTMLOListElement.cpp:
(WebCore::HTMLOListElement::setStartForBindings): Deleted.
* Source/WebCore/html/HTMLOListElement.h: Removed setStartForBindings.
* Source/WebCore/html/HTMLOListElement.idl: Use ReflectSetter.

* Source/WebCore/html/HTMLOptionElement.cpp:
(WebCore::HTMLOptionElement::createForLegacyFactoryFunction):
Use setAttributeWithoutSynchronization.
(WebCore::HTMLOptionElement::value const):
(WebCore::HTMLOptionElement::label const):
(WebCore::HTMLOptionElement::displayLabel const):
(WebCore::HTMLOptionElement::collectOptionInnerTextCollapsingWhitespace const):
(WebCore::HTMLOptionElement::setValue): Deleted.
(WebCore::HTMLOptionElement::setLabel): Deleted.
* Source/WebCore/html/HTMLOptionElement.h: Removed setValue and setLabel.
* Source/WebCore/html/HTMLOptionElement.idl: Use ReflectSetter.

* Source/WebCore/html/HTMLProgressElement.cpp:
(WebCore::HTMLProgressElement::setValue): Deleted.
* Source/WebCore/html/HTMLProgressElement.h: Removed setValue.
* Source/WebCore/html/HTMLProgressElement.idl: Use ReflectSetter.

* Source/WebCore/html/HTMLScriptElement.cpp:
(WebCore::HTMLScriptElement::setCrossOrigin): Deleted.
(WebCore::HTMLScriptElement::setReferrerPolicyForBindings): Deleted.
(WebCore::HTMLScriptElement::setFetchPriorityForBindings): Deleted.
* Source/WebCore/html/HTMLScriptElement.h: Removed setCrossOrigin,
setReferrerPolicyForBindings, and setFetchPriorityForBindings.
* Source/WebCore/html/HTMLScriptElement.idl: Use ReflectSetter.

* Source/WebCore/html/HTMLSelectElement.idl: Use ReflectSetter.

* Source/WebCore/html/HTMLTableCellElement.cpp:
(WebCore::HTMLTableCellElement::setScope): Deleted.
* Source/WebCore/html/HTMLTableCellElement.h: Removed setScope.
* Source/WebCore/html/HTMLTableCellElement.idl: Use ReflectSetter.

* Source/WebCore/html/HTMLTemplateElement.cpp:
(WebCore::HTMLTemplateElement::setShadowRootMode): Deleted.
* Source/WebCore/html/HTMLTemplateElement.h: Removed setShadowRootMode.
* Source/WebCore/html/HTMLTemplateElement.idl: Use ReflectSetter.

* Source/WebCore/html/HTMLTextAreaElement.idl: Use ReflectSetter.

* Source/WebCore/html/HTMLTrackElement.cpp:
(WebCore::HTMLTrackElement::HTMLTrackElement): Replaced pointless code that
called attributeWithoutSynchronization in the constructor. There are no
attributes yet at construction time, so this is just a slow way to get what
is always a null string.
(WebCore::HTMLTrackElement::setKind): Deleted.
(WebCore::HTMLTrackElement::srclang const): Deleted.
(WebCore::HTMLTrackElement::label const): Deleted.
* Source/WebCore/html/HTMLTrackElement.h: Deleted setKind, srclang, and label.
* Source/WebCore/html/HTMLTrackElement.idl: Use ReflectSetter.

* Source/WebCore/html/HTMLVideoElement.cpp:
(WebCore::HTMLVideoElement::setWebkitWirelessVideoPlaybackDisabled): Deleted.
* Source/WebCore/html/HTMLVideoElement.h: Removed setWebkitWirelessVideoPlaybackDisabled.
* Source/WebCore/html/HTMLVideoElement.idl: Use ReflectSetter.

* Source/WebCore/html/ImageDocument.cpp:
(WebCore::ImageDocument::createDocumentStructure): Use setAttributeWithoutSynchronization.
(WebCore::ImageDocument::resizeImageToFit): Use setIntegralAttribute.
(WebCore::ImageDocument::restoreImageSize): Use setIntegralAttribute.

* Source/WebCore/html/InputType.cpp:
(WebCore::InputType::saveFormControlState const): Use attributeWithoutSynchronization.

* Source/WebCore/html/PermissionsPolicy.cpp:
(WebCore::PermissionsPolicy::processPermissionsPolicyAttribute): Use
hasAttributeWithoutSynchronization.

* Source/WebCore/html/PopoverInvokerElement.idl: Use ReflectSetter.

* Source/WebCore/html/shadow/DateTimeNumericFieldElement.cpp:
(WebCore::DateTimeNumericFieldElement::setARIAValueAttributesWithInteger):
Use a single call to AtomString::number for better efficiency.

* Source/WebCore/loader/ImageLoader.cpp:
(WebCore::ImageLoader::updateFromElement): Use getURLAttribute.

* Source/WebCore/mathml/MathMLMencloseElement.cpp:
(WebCore::MathMLMencloseElement::parseNotationAttribute): Use
hasAttributeWithoutSynchronization.

* Source/WebCore/mathml/MathMLSelectElement.cpp:
(WebCore::MathMLSelectElement::getSelectedActionChildAndIndex): Use
integralAttribute.

* Source/WebCore/page/DragController.cpp:
(WebCore::documentFragmentFromDragData): Use setAttributeWithoutSynchronization.

* Source/WebCore/page/text-extraction/TextExtraction.cpp:
(WebCore::TextExtraction::extractItemData): Use getURLAttribute.

* Source/WebCore/rendering/HitTestResult.cpp:
(WebCore::HitTestResult::altDisplayString const): Use attributeWithoutSynchronization.

* Source/WebKitLegacy/mac/DOM/DOMHTMLBaseElement.mm:
(-[DOMHTMLBaseElement setHref:]): Use setAttributeWithoutSynchronization.
* Source/WebKitLegacy/mac/DOM/DOMHTMLButtonElement.mm:
(-[DOMHTMLButtonElement setType:]): Use setAttributeWithoutSynchronization.
* Source/WebKitLegacy/mac/DOM/DOMHTMLElement.mm:
(-[DOMHTMLElement setDir:]): Use setAttributeWithoutSynchronization.
(-[DOMHTMLElement setAutocapitalize:]): Use setAttributeWithoutSynchronization.
* Source/WebKitLegacy/mac/DOM/DOMHTMLEmbedElement.mm:
(-[DOMHTMLEmbedElement height]): Use integralAttribute.
(-[DOMHTMLEmbedElement width]): Use integralAttribute.
* Source/WebKitLegacy/mac/DOM/DOMHTMLFormElement.mm:
(-[DOMHTMLFormElement setAutocomplete:]): Use setAttributeWithoutSynchronization.
(-[DOMHTMLFormElement setEnctype:]): Use setAttributeWithoutSynchronization.
(-[DOMHTMLFormElement setEncoding:]): Use setAttributeWithoutSynchronization.
(-[DOMHTMLFormElement setMethod:]): Use setAttributeWithoutSynchronization.
* Source/WebKitLegacy/mac/DOM/DOMHTMLImageElement.mm:
(-[DOMHTMLImageElement setCrossOrigin:]): Use setAttributeWithoutSynchronization.
(-[DOMHTMLImageElement setHeight:]): Use setAttributeWithoutSynchronization.
(-[DOMHTMLImageElement hspace]): Use integralAttribute.
(-[DOMHTMLImageElement vspace]): Use integralAttribute.
(-[DOMHTMLImageElement setWidth:]): Use setIntegralAttribute.
(-[DOMHTMLImageElement altDisplayString]): Use attributeWithoutSynchronization.
* Source/WebKitLegacy/mac/DOM/DOMHTMLInputElement.mm:
(-[DOMHTMLInputElement setAutocomplete:]): Use setAttributeWithoutSynchronization.
(-[DOMHTMLInputElement setFormAction:]): Use setAttributeWithoutSynchronization.
(-[DOMHTMLInputElement setFormEnctype:]): Use setAttributeWithoutSynchronization.
(-[DOMHTMLInputElement setFormMethod:]): Use setAttributeWithoutSynchronization.
(-[DOMHTMLInputElement setHeight:]): Use setUnsignedIntegralAttribute.
(-[DOMHTMLInputElement max]): Use attributeWithoutSynchronization.
(-[DOMHTMLInputElement min]): Use attributeWithoutSynchronization.
(-[DOMHTMLInputElement pattern]): Use attributeWithoutSynchronization.
(-[DOMHTMLInputElement placeholder]): Use attributeWithoutSynchronization.
(-[DOMHTMLInputElement step]): Use attributeWithoutSynchronization.
(-[DOMHTMLInputElement setType:]): Use setAttributeWithoutSynchronization.
(-[DOMHTMLInputElement defaultValue]): Use attributeWithoutSynchronization.
(-[DOMHTMLInputElement setDefaultValue:]): Use setAttributeWithoutSynchronization.
(-[DOMHTMLInputElement setWidth:]): Use setUnsignedIntegralAttribute.
(-[DOMHTMLInputElement altDisplayString]): Use attributeWithoutSynchronization.
* Source/WebKitLegacy/mac/DOM/DOMHTMLLIElement.mm:
(-[DOMHTMLLIElement value]): Use integralAttribute.
* Source/WebKitLegacy/mac/DOM/DOMHTMLLinkElement.mm:
(-[DOMHTMLLinkElement setCrossOrigin:]): Use setAttributeWithoutSynchronization.
* Source/WebKitLegacy/mac/DOM/DOMHTMLMarqueeElement.mm:
(-[DOMHTMLMarqueeElement hspace]): Use unsignedIntegralAttribute.
(-[DOMHTMLMarqueeElement vspace]): Use unsignedIntegralAttribute.
* Source/WebKitLegacy/mac/DOM/DOMHTMLMediaElement.mm:
(-[DOMHTMLMediaElement setCrossOrigin:]): Use setAttributeWithoutSynchronization.
* Source/WebKitLegacy/mac/DOM/DOMHTMLOListElement.mm:
(-[DOMHTMLOListElement setStart:]): Use setIntegralAttribute.
* Source/WebKitLegacy/mac/DOM/DOMHTMLObjectElement.mm:
(-[DOMHTMLObjectElement hspace]): Use integralAttribute.
(-[DOMHTMLObjectElement vspace]): Use integralAttribute.
* Source/WebKitLegacy/mac/DOM/DOMHTMLOptionElement.mm:
(-[DOMHTMLOptionElement setLabel:]): Use setAttributeWithoutSynchronization.
(-[DOMHTMLOptionElement setValue:]): Use setAttributeWithoutSynchronization.
* Source/WebKitLegacy/mac/DOM/DOMHTMLPreElement.mm:
(-[DOMHTMLPreElement width]): Use integralAttribute.
* Source/WebKitLegacy/mac/DOM/DOMHTMLScriptElement.mm:
(-[DOMHTMLScriptElement htmlFor]): Use attributeWithoutSynchronization.
(-[DOMHTMLScriptElement event]): Use attributeWithoutSynchronization.
(-[DOMHTMLScriptElement charset]): Use attributeWithoutSynchronization.
(-[DOMHTMLScriptElement type]): Use attributeWithoutSynchronization.
(-[DOMHTMLScriptElement setCrossOrigin:]): Use setAttributeWithoutSynchronization.
(-[DOMHTMLScriptElement nonce]): Use attributeWithoutSynchronization.
* Source/WebKitLegacy/mac/DOM/DOMHTMLTableCellElement.mm:
(-[DOMHTMLTableCellElement setScope:]): Use setAttributeWithoutSynchronization.
* Source/WebKitLegacy/mac/DOM/DOMHTMLTextAreaElement.mm:
(-[DOMHTMLTextAreaElement setAutocomplete:]): Use setAttributeWithoutSynchronization.
* Source/WebKitLegacy/mac/DOM/DOMHTMLVideoElement.mm:
(-[DOMHTMLVideoElement width]): Use unsignedIntegralAttribute.
(-[DOMHTMLVideoElement height]): Use unsignedIntegralAttribute.

Canonical link: https://commits.webkit.org/296427@main
